rq_scheduler/scheduler.py

```diff
@@ -28,6 +28,18 @@ def __init__(self, queue_name='default', interval=60, connection=None):
         self.log = logger
         self._lock_acquired = False
 
+    def __enter__(self):
+        while True:
+            try:
+                self.register_birth()
+                return self
+            except ValueError:  # assume register_birth failed
+                self.log.info('Waiting for registering birth...')
+                time.sleep(self._interval)
```

Comment:
I think we should let this exception be raised and let the user handle it. I'm not comfortable with implicitly calling `sleep()` which will block code execution.